src/megatron/bridge/training/train.py

```diff
@@ -660,8 +661,9 @@ def disable_forward_pre_hook(model: list[DDP], param_sync: bool = True) -> None:
         param_sync: Whether to synchronize parameters across model chunks
     """
     for model_chunk in model:
-        assert isinstance(model_chunk, DDP)
-        model_chunk.disable_forward_pre_hook(param_sync=param_sync)
+        # only works for DDP, Megatron FSDP or FSDP2 does not have this method
+        if isinstance(model_chunk, DDP):
+            model_chunk.disable_forward_pre_hook(param_sync=param_sync)
```

Comment:
we shouldn't need these changes here. we should change the condition in L188 to check that megatron FSDP is not used and (distributed optimizer + overlap param gather are used)